### PR TITLE
feat: Use semantic versioning for Sentry releases

### PR DESCRIPTION
**Pull request by billyvg**
Instead of using the commit SHA for our Sentry releases, use a more semantic version: CalVer+<build SHA>

This also splits up the backend/frontend Sentry releases where previously they were in the same release (which implies they are always deployed together). This will be needed for independent frontend deploys.

### CODE REVIEW

File: src/sentry/__init__.py [hunk 1]
```diff
@@ -52,6 +59,9 @@ def is_docker():
 __build__ = get_revision()
 __docformat__ = "restructuredtext en"
 
+_build_or_empty = "" if __build__ is None else f"+{get_short_revision(__build__)}"
+__full_version__ = f"{__version__}{_build_or_empty}"
```
Comment (EvanPurkhiser) [inline src/sentry/__init__.py:63]: Should we toss some comments above these?

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/__init__.py (hunk 1 above)
Comment (joshuarli) [inline src/sentry/__init__.py:63]: You can just reduce this to `__full_version__ = __version__ if __build__ is None else f"{__version__}+{__build__[:12]}"`.

Also I don't think `__full_version__`  is a good name, it's confusing with the short sha. Maybe `__release_version__` and include a little comment about what it's for.

Comment (joshuarli) [reply src/sentry/__init__.py:63]: Oh or `__semantic_version__`.

Comment (joshuarli) [reply src/sentry/__init__.py:63]: Ehh, I see get_short_revision being used in https://github.com/getsentry/getsentry/pull/6265 and I kind of think `__full_version__` is ok now.

**Review by joshuarli — APPROVED** [review]

File: src/sentry/__init__.py (hunk 1 above)
Comment (billyvg) [reply src/sentry/__init__.py:63]: I think release/semantic might be better. Will think about it tomorrow and add comments.

**Pull request merged**